packages/framework/package.json

```diff
@@ -38,7 +38,7 @@
     "build:watch": "tsup --watch",
     "postbuild": "pnpm run check:exports && pnpm check:circulars",
     "check:exports": "attw --pack .",
-    "check:circulars": "madge --circular --extensions ts ./src",
+    "check:circulars": "madge --circular --extensions ts --exclude ../../shared ./src",
```

Comment:
@SokratisVidros is seems that Framework is still detecting circular dependency issues from Framework, even though no issues are traced back to Framework (see logs below in collapsible).

Furthermore it appears the PR removing shared circulars had a [run](https://github.com/novuhq/novu/actions/runs/12081806479/job/33691700651#step:5:59) pulled from [local cache](https://cloud.nx.app/runs/DEHUVDyD6H/task/%40novu%2Fframework%3Abuild) on CI, so this removal was never tested. I am reverting for now as it's a false-positive.

<details><summary>Details</summary>
<p>

```bash
1) ../../shared/dist/cjs/dto/index.d.ts > ../../shared/dist/cjs/dto/events/index.d.ts > ../../shared/dist/cjs/dto/events/event.interface.d.ts > ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/clerk/index.d.ts > ../../shared/dist/cjs/types/clerk/types.d.ts
2) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/clerk/index.d.ts > ../../shared/dist/cjs/types/clerk/types.d.ts > ../../shared/dist/cjs/entities/user/index.d.ts > ../../shared/dist/cjs/entities/user/user.interface.d.ts
3) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/handlebar-helpers/index.d.ts > ../../shared/dist/cjs/consts/handlebar-helpers/getTemplateVariables.d.ts
4) ../../shared/dist/cjs/dto/index.d.ts > ../../shared/dist/cjs/dto/events/index.d.ts > ../../shared/dist/cjs/dto/events/event.interface.d.ts > ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/message-template/index.d.ts > ../../shared/dist/cjs/entities/message-template/message-template.interface.d.ts
5) ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/message-template/index.d.ts > ../../shared/dist/cjs/entities/message-template/message-template.interface.d.ts
6) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/message-template/index.d.ts > ../../shared/dist/cjs/entities/message-template/message-template.interface.d.ts
7) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/step/index.d.ts
8) ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/subscriber-preference/index.d.ts > ../../shared/dist/cjs/entities/subscriber-preference/subscriber-preference.interface.d.ts
9) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/entities/subscriber-preference/index.d.ts > ../../shared/dist/cjs/entities/subscriber-preference/subscriber-preference.interface.d.ts
10) ../../shared/dist/cjs/clients/index.d.ts > ../../shared/dist/cjs/clients/workflows-client.d.ts > ../../shared/dist/cjs/dto/index.d.ts > ../../shared/dist/cjs/dto/events/index.d.ts > ../../shared/dist/cjs/dto/events/event.interface.d.ts > ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
11) ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
12) ../../shared/dist/cjs/dto/index.d.ts > ../../shared/dist/cjs/dto/events/index.d.ts > ../../shared/dist/cjs/dto/events/event.interface.d.ts > ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
13) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/environment/index.d.ts > ../../shared/dist/cjs/entities/environment/environment.interface.d.ts
14) ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/job/index.d.ts > ../../shared/dist/cjs/entities/job/job.interface.d.ts
15) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/job/index.d.ts > ../../shared/dist/cjs/entities/job/job.interface.d.ts
16) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/layout/index.d.ts > ../../shared/dist/cjs/entities/layout/layout.interface.d.ts
17) ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
18) ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
19) ../../shared/dist/cjs/dto/index.d.ts > ../../shared/dist/cjs/dto/events/index.d.ts > ../../shared/dist/cjs/dto/events/event.interface.d.ts > ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/organization/index.d.ts > ../../shared/dist/cjs/entities/organization/organization.interface.d.ts
20) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/organization/index.d.ts > ../../shared/dist/cjs/entities/organization/organization.interface.d.ts
21) ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/subscriber/index.d.ts > ../../shared/dist/cjs/entities/subscriber/subscriber.interface.d.ts
22) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/subscriber/index.d.ts > ../../shared/dist/cjs/entities/subscriber/subscriber.interface.d.ts
23) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/tenant/index.d.ts > ../../shared/dist/cjs/entities/tenant/tenant.interface.d.ts
24) ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/workflow-override/index.d.ts > ../../shared/dist/cjs/entities/workflow-override/workflow-override.interface.d.ts
25) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/entities/workflow-override/index.d.ts > ../../shared/dist/cjs/entities/workflow-override/workflow-override.interface.d.ts
26) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts
27) ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/utils/index.d.ts > ../../shared/dist/cjs/utils/bridge.utils.d.ts
28) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/utils/index.d.ts > ../../shared/dist/cjs/utils/bridge.utils.d.ts
29) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/utils/index.d.ts > ../../shared/dist/cjs/utils/buildWorkflowPreferences.d.ts
30) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts > ../../shared/dist/cjs/index.d.ts > ../../shared/dist/cjs/utils/index.d.ts > ../../shared/dist/cjs/utils/checkIsResponseError.d.ts
31) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts > ../../shared/dist/cjs/entities/notification-template/index.d.ts > ../../shared/dist/cjs/entities/notification-template/notification-template.interface.d.ts
32) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/index.d.ts > ../../shared/dist/cjs/consts/notification-item-buttons/notificationItemButton.d.ts > ../../shared/dist/cjs/entities/messages/index.d.ts > ../../shared/dist/cjs/entities/messages/messages.interface.d.ts
33) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/preferences/index.d.ts > ../../shared/dist/cjs/consts/preferences/preferences.const.d.ts
34) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/productFeatureEnabledForServiceLevel.d.ts
35) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/providers/index.d.ts > ../../shared/dist/cjs/consts/providers/credentials/index.d.ts > ../../shared/dist/cjs/consts/providers/credentials/provider-credentials.d.ts > ../../shared/dist/cjs/consts/providers/provider.interface.d.ts
36) ../../shared/dist/cjs/types/index.d.ts > ../../shared/dist/cjs/types/subscriber/index.d.ts > ../../shared/dist/cjs/consts/index.d.ts > ../../shared/dist/cjs/consts/rate-limiting/index.d.ts > ../../shared/dist/cjs/consts/rate-limiting/apiRateLimits.d.ts
37) ../../shared/dist/cjs/dto/workflows/workflow-commons-fields.d.ts > ../../shared/dist/cjs/dto/workflows/workflow-response-dto.d.ts > ../../shared/dist/cjs/dto/workflows/create-workflow-dto.d.ts
38) ../../shared/dist/cjs/dto/workflows/update-workflow-dto.d.ts > ../../shared/dist/cjs/dto/workflows/workflow-commons-fields.d.ts > ../../shared/dist/cjs/dto/workflows/workflow-response-dto.d.ts
39) ../../shared/dist/cjs/dto/workflows/workflow-commons-fields.d.ts > ../../shared/dist/cjs/dto/workflows/workflow-response-dto.d.ts
```

</p>
</details>